src/core/Dropdown/Dropdown/Dropdown.tsx

```diff
@@ -180,11 +186,241 @@ class BaseDropdown extends Component<DropdownProps> {
   ) {
     const { value } = nextProps;
     if ('value' in nextProps && value !== prevState.selectedValue) {
-      return { selectedValue: value };
+      return {
+        selectedValue: value,
+        selectedValueText: BaseDropdown.parseSelectedValueText(
+          value,
+          nextProps.children,
+        ),
+      };
+    }
+    return null;
+  }
+
+  static parseSelectedValueText(
+    selectedValue: string | undefined,
+    children:
+      | Array<ReactElement<DropdownItemProps>>
+      | ReactElement<DropdownItemProps>
+      | undefined,
+  ): string | undefined {
+    if (selectedValue === undefined || children === undefined) return undefined;
+
+    if (Array.isArray(children)) {
+      for (let index = 0; index < children.length; index += 1) {
+        const element = children[index];
+        if (element.props.value === selectedValue) {
+          return getRecursiveChildText(element);
+        }
+      }
+    } else {
+      return getRecursiveChildText(children);
+    }
+  }
+
+  private isOutsideClick(event: MouseEvent) {
+    return (
+      !!this.buttonRef &&
+      (this.buttonRef.current as Node).contains(event.target as Node)
+    );
+  }
+
+  private handleItemSelection(itemValue: string) {
+    if (!!this.props.onChange) {
+      this.props.onChange(itemValue);
+    }
+    this.setState({
+      selectedValue: itemValue,
+      selectedValueText: BaseDropdown.parseSelectedValueText(
+        itemValue,
+        this.props.children,
+      ),
+      showPopover: false,
+      focusedDescendantId: itemValue,
+    });
+    /**
+     * Timeout is used here to ensure screen reader reads the
+     * currently selected option when focusing back to the button
+     */
+    setTimeout(() => {
+      this.buttonRef.current?.focus();
+    }, 200);
```

Comment:
```suggestion
  this.setState({ showPopover: false })
    }, 200);
```

Reply:
Setting showPopover false after focusing to button should prevent NVDA from starting to read out the document title during the timeout